### PR TITLE
Left menu adaptation proposal

### PR DESCRIPTION
**Pull request by cmangeat**

### CODE REVIEW

File: c2cgeoform/routes.py [hunk 1]
```diff
@@ -13,19 +11,25 @@ def table_pregenerator(request, elements, kw):
...
         table['key'] = key
-        table['name'] = mapper.__tablename__
-        table['title'] = ca_config.get('title', table['name'])
-        table['plural'] = ca_config.get('plural', table['title'])
+        table['title'] = ca_config.get('title', mapper.__tablename__)
+        table['plural'] = ca_config.get('plural', mapper.__tablename__)
         tables.append(table)
 
-    table_regex = '[{}]'.format('|'.join([table['key'] for table in tables]))
+    table_regex = '({})'.format(')|('.join([table['key'] for table in tables]))
```
Comment (arnaud-morvan) [inline c2cgeoform/routes.py]: I would prefer '|'.join('({})'.format([table['key']) for table in tables])
for understanding

Comment (sbrunner) [reply c2cgeoform/routes.py]: +1

File: c2cgeoform/routes.py [hunk 2]
```diff
@@ -1,6 +1,4 @@
-
-
-tables=[]
```
Comment (arnaud-morvan) [inline c2cgeoform/routes.py]: If tables is not global, you lose the possibility to call register_models many times

File: c2cgeoform/routes.py [hunk 3]
```diff
@@ -13,19 +11,25 @@ def table_pregenerator(request, elements, kw):
...
     rec_with_pregenerator('c2cgeoform_index', '{}/'.format(base_route))
     rec_with_pregenerator('c2cgeoform_grid', '{}/grid.json'.format(base_route))
-    rec_with_pregenerator('c2cgeoform_action', '{}/{{id}}/{{action}}'.format(base_route))
+    rec_with_pregenerator('c2cgeoform_action', '{}/{{id}}/{{action}}'
+                          .format(base_route))
+
+    def add_global(event):
+        event['tables'] = tables
+
+    config.add_subscriber(add_global, iface=BeforeRender)
```
Comment (arnaud-morvan) [inline c2cgeoform/routes.py]: If register_models is called many times, you will register the subscriber many times.

Comment (cmangeat) [reply c2cgeoform/routes.py]: You cant anyway call register_models manytimes, as it is not possible to register the same route with differents pattern many times.

Comment (arnaud-morvan) [reply c2cgeoform/routes.py]: Are you sure about this ? What append if you do it ?

Comment (arnaud-morvan) [reply c2cgeoform/routes.py]: I suppose we should add some tests if we want to keep this as a functionnality.

Comment (cmangeat) [reply c2cgeoform/routes.py]: pyramid configuration conflict exception or something tasting so

Comment (arnaud-morvan) [reply c2cgeoform/routes.py]: Ok so everything seems good on this point

**Pull request merged**